### PR TITLE
Conversion of OOXML to CiceroMark and vice versa

### PR DESCRIPTION
**Pull request by algomaster99**
# Issue #247 
Add package for converting `OOXML` <-> `CiceroMark`

### Changes
- [x] initialize package
- [ ] add function for converting OOXML to CiceroMark with proper tests
- [ ] add function for converting CiceroMark to OOXML with proper tests

### CODE REVIEW

**Comment by algomaster99** [comment]
Oh, I just thought it was supposed to be a different package. One reason I can think of is that `markdown-docx` uses a **path to a DOCX file** which I _think_ will install dependencies which might only work on Node and not on the browser. `Mammoth` works on the web too though.

**Comment by jeromesimeon** [comment]
> Oh, I just thought it was supposed to be a different package. One reason I can think of is that `markdown-docx` uses a **path to a DOCX file** which I _think_ will install dependencies which might only work on Node and not on the browser. `Mammoth` works on the web too though.

I think it would make sense for all transforms to work on input data in memory and not depend on the file system? Maybe I'm missing something there.

Also, it would help to figure out if there are any differences between ooxml or docx before creating a whole new package.

@DianaLease @irmerk @dselman help here please?

**Comment by dselman** [comment]
> Why is this a different package from `docx` ?

I think making this `OOXML` is more descriptive - as technically that is the (open) data format, vs docx which is the Microsoft proprietary file wrapper.

**Comment by algomaster99** [comment]
@dselman But we will only be able to transform OOXML from Word Documents to CiceroMark. Other OOXML (for eg. from PowerPoint, Excel, etc.) will fail as it makes no sense.

**Comment by jeromesimeon** [comment]
> > Why is this a different package from `docx` ?
> 
> I think making this `OOXML` is more descriptive - as technically that is the (open) data format, vs docx which is the Microsoft proprietary file wrapper.

Sounds like an argument for renaming the current package, not creating a new one?

Also, maybe I'm wrong but isn't a docx containing several xml files? (some of which may or may not be ooxml?).

**Comment by algomaster99** [comment]
> Also, maybe I'm wrong but isn't a docx containing several xml files?

@jeromesimeon OOXML file is just a zipped collection of all the XML files the Word contains as written [here](http://officeopenxml.com/).

**Comment by irmerk** [comment]
After discussing on a call, I think it would likely make the most sense to keep the code in `markdown-docx` instead of making a new package. In the future we can change the name if that makes sense... But based on @algomaster99's findings it seems like `docx` is the most descriptive at the moment.

This will get rid of `mammoth` and replace the [current functionality][code] with the implementation Aman creates. 

[code]: https://github.com/accordproject/markdown-transform/blob/master/packages/markdown-docx/src/DocxTransformer.js#L42-L45

**Comment by irmerk** [comment]
@algomaster99 I think we wanted to keep it `markdown-docx` instead of `markdown-ooxml`

File: packages/markdown-ooxml/src/OoxmlTransformer.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+     * @returns {string} the name/id of the variable
+     */
+    getId(elements) {
+        const variableProperties = elements[0]
+        for (const property of variableProperties.elements) {
+            if (property.name === 'w:tag') {
+                return property.attributes['w:val'];
+            }
+        }
+    }
```
Comment (irmerk) [inline packages/markdown-ooxml/src/OoxmlTransformer.js:38]: Is there a case where `getId()` will not return anything, and you'd want to have a base case after the for loop?

**Comment by algomaster99** [comment]
@irmerk cool, I will try to integrate it there. Once I will do that, I will close this PR.

File: packages/markdown-ooxml/src/OoxmlTransformer.js (hunk 1 above)
Comment (algomaster99) [reply packages/markdown-ooxml/src/OoxmlTransformer.js:38]: @irmerk I think every variable will have a corresponding ID so it ought to return something. And I am only calling this function when I parse `w:sdt` which are node types for content controls.

**Pull request closed without merging**